### PR TITLE
Changed needed for Zephyr

### PR DESCRIPTION
**Pull request by jukkar**
Initial changes needed for compiling with Zephyr.
Corresponding glue code in https://github.com/zephyrproject-rtos/zephyr/pull/64168 will be sent next.

This does not yet contain driver integration code as I will sent it after this one is merged.

### CODE REVIEW

**Comment by jukkar** [comment]
cc: @krish2718

**Comment by jukkar** [comment]
Related Zephyr PR is https://github.com/zephyrproject-rtos/zephyr/pull/64168

**Review by krish2718 — APPROVED** [review]

File: src/utils/common.c [hunk 1]
```diff
@@ -1302,3 +1302,13 @@ void forced_memzero(void *ptr, size_t len)
 	if (len)
 		forced_memzero_val = ((u8 *) ptr)[0];
 }
+
+#ifdef CONFIG_ZEPHYR
+#include <zephyr/net/net_ip.h>
+extern char *inet_ntoa(struct in_addr in)
+{
+	char addr[NET_IPV4_ADDR_LEN];
```
Comment (rlubos) [inline src/utils/common.c:1310]: Shouldn't this be `static`, given the pointer is returned?

File: src/utils/os_zephyr.c [hunk 2]
```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
```
[the remark this answers is not in the document]
Comment (jukkar) [reply src/utils/os_zephyr.c:2]: Yes we should, I will fix it.

File: src/utils/common.c (hunk 1 above)
Comment (jukkar) [reply src/utils/common.c:1310]: Yes, good point.

**Review by rlubos — APPROVED** [review]

**Pull request merged**